lib/utils.js

```diff
@@ -486,10 +486,11 @@ function prettifyError ({ keyName, lines, eol, ident }) {
  * @param {object} log The log object to be modified.
  * @param {string} property A string identifying the property to be deleted from
  * the log object. Accepts nested properties delimited by a `.`
- * e.g. `'prop1.prop2'`.
+ * Delimiter can be escaped to preserve property names that contain the delimiter.
+ * e.g. `'prop1.prop2'` or `'prop2\.domain\.com.prop2'`
  */
 function deleteLogProperty (log, property) {
-  const props = property.split('.')
+  const props = property.match(/([^\\.]|\\\.)+/g).map((e) => e.split('\\.').join('.'))
```

Comment:
how expensive is this operation?

Reply:
https://jsbench.me/tqkrafwdkf/1

I just performed some benchmarks, this implementation is ~81% slower compared to just a split.

There might be quite a hit on performance with this change, however, this can be improved if we split the key using a loop. Benchmark shows that it is ~36% slower compared to just a split.

Will be pushing the improved version shortly for review.

Reply:
Thank you!